src/importers.nix

```
@@ -18,38 +77,36 @@ let
       ./main/dev.nix
       ./main/os/default.nix
       ```
+
       Example output:
       ```
       {
-      core = ./core;
-      base = base.nix;
-      main = {
-      dev = ./main/dev.nix;
-      os = ./main/os;
-      };
+        core = ./core;
+        base = base.nix;
+        main = {
+          dev = ./main/dev.nix;
+          os = ./main/os;
+        };
       }
       ```
       **/
     dirPath:
     let
-      # Relative paths cause issues, so convert to string immediately
-      dir = toString dirPath;
-
       collect = file: type: {
         name = lib.removeSuffix ".nix" file;
-        value = let path = "${dir}/${file}"; in
+        value = let path = dirPath + "/${file}"; in
           if (type == "regular")
-            || (type == "directory" && builtins.pathExists "${path}/default.nix")
+            || (type == "directory" && builtins.pathExists (path + "/default.nix"))
           then path
           else rakeLeaves path;
       };
 
       files = lib.filterAttrs
         (file: type:
-          # Only include `.nix` files or directories
+          # Only rake `.nix` files or directories
           (type == "regular" && lib.hasSuffix ".nix" file) || (type == "directory")
         )
-        (lib.safeReadDir dir);
+        (builtins.readDir dirPath);
```

```suggestion
        (builtins.readDir (toString dirPath));
```
I really think this is a good idea for safety, I don't believe readDir works well with paths. I think thats the function that creates new nix store paths when you pass it a relative path.